crates/client/rpc/src/methods/read/syncing.rs

```diff
@@ -0,0 +1,99 @@
+use jsonrpsee::core::RpcResult;
+use mc_genesis_data_provider::GenesisProvider;
+use mc_sync::l2::get_highest_block_hash_and_number;
+use mp_hashers::HasherT;
+use pallet_starknet_runtime_api::{ConvertTransactionRuntimeApi, StarknetRuntimeApi};
+use sc_client_api::backend::{Backend, StorageProvider};
+use sc_client_api::BlockBackend;
+use sc_transaction_pool::ChainApi;
+use sc_transaction_pool_api::TransactionPool;
+use sp_api::ProvideRuntimeApi;
+use sp_arithmetic::traits::UniqueSaturatedInto;
+use sp_blockchain::HeaderBackend;
+use sp_runtime::traits::Block as BlockT;
+use starknet_core::types::{SyncStatus, SyncStatusType};
+
+use crate::{madara_backend_client, Starknet};
+
+/// Returns an object about the sync status, or false if the node is not synching
+///
+/// ### Arguments
+///
+/// This function does not take any arguments.
+///
+/// ### Returns
+///
+/// * `Syncing` - An Enum that can either be a `mc_rpc_core::SyncStatus` struct representing the
+///   sync status, or a `Boolean` (`false`) indicating that the node is not currently synchronizing.
+///
+/// This is an asynchronous function due to its reliance on `sync_service.best_seen_block()`,
+/// which potentially involves network communication and processing to determine the best block
+/// seen by the sync service.
+#[allow(unused_variables)]
```

Comment:
Same, why are we ignoring unused variables?

Reply:
This appears to be the case for every RPC method. Please remove this as it risks obfuscating future errors.